File: Move_last_element_to_front_of_a_linked_list/code.cpp
class Solution{
public:
  ListNode *moveToFront(ListNode *head){
    if(head == NULL || head->next == NULL){
      return head;
    }
        else{
      ListNode* ptr = head;
      while(ptr ->next->next != NULL){
        ptr = ptr->next;
      }
      ptr->next->next = head;
      ListNode* ans = ptr->next;
      ptr->next = NULL;
      return ans;
    }
  }
};

File: Move_last_element_to_front_of_a_linked_list/code.java
class Solution {
    public static Node moveToFront(Node head) {
        // code here
        if(head==null || head.next==null) return head;
        Node temp=head,prev=null;
        while(temp.next!=null)
        {
            prev=temp;
            temp=temp.next;
        }
        Node node=new Node(temp.data);
        prev.next=null;
        node.next=head;
        return node;
    }
}
